lib/matplotlib/__init__.py

```diff
@@ -547,6 +548,11 @@ def _create_tmp_config_dir():
     tempdir = os.path.join(tempdir, 'matplotlib-%s' % getpass.getuser())
     os.environ['MPLCONFIGDIR'] = tempdir
 
+    try:
+        mkdirs(tempdir)
+    except OSError:
```

Comment:
cbook.py already has a safe mkdirs(). Unfortunately, Python's mkdirs() has
some nasty race conditions and other problems, so you should use the safe
mkdirs.
On Jul 6, 2015 12:48 PM, "Thomas A Caswell" <notifications@github.com>
wrote:

> In lib/matplotlib/__init__.py
> <https://github.com/matplotlib/matplotlib/pull/4591#discussion_r33955250>:
>
> > @@ -547,6 +548,11 @@ def _create_tmp_config_dir():
> >      tempdir = os.path.join(tempdir, 'matplotlib-%s' % getpass.getuser())
> >      os.environ['MPLCONFIGDIR'] = tempdir
> >
> > +    try:
> > +        mkdirs(tempdir)
> > +    except OSError:
>
> I would not eat the exception here, this leaves the user in the same
> situation as before if this fails.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4591/files#r33955250>.
>